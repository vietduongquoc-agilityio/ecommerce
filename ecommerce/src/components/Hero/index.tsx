"use client";

import hero from "@/assets/Images/Hero.png";
import Image from "next/image";
import { Button, Text } from "@radix-ui/themes";
import { colors, fonts } from "@/themes";

const Hero = () => {
  return (
    <section
      style={{
        position: "relative",
        width: "100%",
        height: "auto",
      }}
    >
      <div
        style={{
          display: "flex",
          justifyContent: "center",
          flexDirection: "column",
          background: colors.primary.light,
          width: "645px",
          padding: "62px 43px 37px 39px",
          position: "absolute",
          top: "155px",
          right: "58px",
          height: "445px",
          borderRadius: fonts.borderRadius.sm,
        }}
      >
        <Text
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.semiBold,
            color: colors.secondary.dark,
          }}
        >
          New Arrival
        </Text>
        <Text
          style={{
            fontSize: fonts.size.xxxl,
            fontWeight: fonts.weight.bold,
            color: colors.primary.yellow,
            width: "400px",
          }}
        >
          Discover Our New Collection
        </Text>
        <Text
          style={{
            fontSize: fonts.size.sm,
            fontWeight: fonts.weight.medium,
            color: colors.secondary.dark,
            width: "546px",
            marginBottom: "45px",
          }}
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut elit
          tellus, luctus nec ullamcorper mattis.
        </Text>
        <Button
          style={{
            padding: "35px 72px",
            background: colors.primary.yellow,
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.bold,
            width: "222px",
            borderRadius: "0",
            cursor: "pointer",
          }}
        >
          BY NOW
        </Button>
      </div>
      <Image
        src={hero}
        alt="hero"
        style={{
          width: "100%",
          height: "auto",
          display: "block",
        }}
      />
    </section>
  );
};

export default Hero;
