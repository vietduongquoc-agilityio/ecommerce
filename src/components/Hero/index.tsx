import Image from "next/image";
import { Text } from "@radix-ui/themes";
import { colors, fonts } from "@/themes";

// Images
import hero from "@/assets/Images/Hero.png";

// Components
import { Button } from "..";

const Hero = () => {
  return (
    <section
      style={{
        position: "relative",
        width: "100%",
        height: "auto",
      }}
    >
      <div
        style={{
          display: "flex",
          justifyContent: "center",
          flexDirection: "column",
          background: colors.primary.light,
          width: "645px",
          padding: "62px 43px 37px 39px",
          position: "absolute",
          top: "155px",
          right: "58px",
          height: "445px",
          borderRadius: fonts.borderRadius.sm,
        }}
      >
        <Text
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.semiBold,
            color: colors.secondary.dark,
          }}
        >
          New Arrival
        </Text>
        <Text
          style={{
            fontSize: fonts.size.xxxl,
            fontWeight: fonts.weight.bold,
            color: colors.primary.yellow,
            width: "400px",
          }}
        >
          Discover Our New Collection
        </Text>
        <Text
          style={{
            fontSize: fonts.size.sm,
            fontWeight: fonts.weight.medium,
            color: colors.secondary.dark,
            width: "546px",
            marginBottom: "45px",
          }}
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut elit
          tellus, luctus nec ullamcorper mattis.
        </Text>
        <div
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.bold,
            width: "225px",
            height: "75px",
          }}
        >
          <Button variant="primary" fullWidth fullHeight>
            BY NOW
          </Button>
        </div>
      </div>
      <Image
        src={hero}
        alt="hero"
        style={{
          width: "100%",
          height: "auto",
          display: "block",
        }}
      />
    </section>
  );
};

export default Hero;
