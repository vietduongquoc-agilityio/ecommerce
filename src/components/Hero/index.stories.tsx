import { Meta, StoryObj } from "@storybook/react";
import Hero from "./";

const meta: Meta<typeof Hero> = {
  title: "Components/Hero",
  component: Hero,
  tags: ['autodocs'],
  parameters: {
    layout: "fullscreen",
  },

};

export default meta;

type Story = StoryObj<typeof Hero>;

export const Default: Story = {};
