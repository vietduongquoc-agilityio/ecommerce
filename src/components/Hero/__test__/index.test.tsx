import { render } from "@testing-library/react";
import Hero from "../index";
import { useRouter } from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

const mockPush = jest.fn();
describe("Hero Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();

    (useRouter as jest.Mock).mockReturnValue({ push: mockPush });
  });
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Hero />);
    expect(asFragment()).toMatchSnapshot();
  });
});
