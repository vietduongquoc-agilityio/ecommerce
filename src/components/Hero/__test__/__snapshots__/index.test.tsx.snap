// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <section
    style="position: relative; width: 100%; margin-bottom: 50px; height: auto; object-fit: cover;"
  >
    <div
      style="display: flex; justify-content: center; flex-direction: column; background: rgb(255, 243, 227); width: 645px; padding: 62px 43px 37px 39px; position: absolute; top: 155px; right: 58px; height: 445px; border-radius: 10px;"
    >
      <span
        class="rt-Text"
        style="font-size: 16px; font-weight: 600; color: rgb(51, 51, 51);"
      >
        New Arrival
      </span>
      <span
        class="rt-Text"
        style="font-size: 52px; font-weight: 700; color: rgb(184, 142, 47); width: 400px;"
      >
        Discover Our New Collection
      </span>
      <span
        class="rt-Text"
        style="font-size: 18px; font-weight: 500; color: rgb(51, 51, 51); width: 546px; margin-bottom: 45px;"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut elit tellus, luctus nec ullamcorper mattis.
      </span>
      <div
        style="font-size: 16px; font-weight: 700; width: 225px; height: 75px;"
      >
        <button
          style="padding: 12px 52px; font-size: 18px; font-weight: 600; text-align: center; width: 100%; height: 100%; cursor: pointer; transition: all 0.3s ease; background-color: rgb(184, 142, 47); color: rgb(255, 255, 255);"
        >
          BUY NOW
        </button>
      </div>
    </div>
    <img
      alt="hero"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent; width: 100%; display: block;"
      width="40"
    />
  </section>
</DocumentFragment>
`;
